siatest/testgroup.go

```diff
@@ -293,40 +292,46 @@ func setRenterAllowances(renters map[*TestNode]struct{}) error {
 }
 
 // synchronizationCheck makes sure that all the nodes are synced and follow the
-func synchronizationCheck(miner *TestNode, nodes map[*TestNode]struct{}) error {
-	mcg, err := miner.ConsensusGet()
+func synchronizationCheck(nodes map[*TestNode]struct{}) error {
+	// Get node with longest chain.
+	var longestChainNode *TestNode
+	var longestChain types.BlockHeight
+	for n := range nodes {
+		ncg, err := n.ConsensusGet()
+		if err != nil {
+			return err
+		}
+		if ncg.Height > longestChain {
+			longestChain = ncg.Height
+			longestChainNode = n
+		}
+	}
+	lcg, err := longestChainNode.ConsensusGet()
 	if err != nil {
 		return err
 	}
-	// Loop until all the blocks have the same CurrentBlock. If we need to mine
-	// a new block in between we need to repeat the check until no block was
-	// mined.
-	for node := range nodes {
+	// Loop until all the blocks have the same CurrentBlock.
+	for n := range nodes {
 		err := Retry(600, 100*time.Millisecond, func() error {
-			ncg, err := node.ConsensusGet()
+			ncg, err := n.ConsensusGet()
 			if err != nil {
 				return err
 			}
 			// If the CurrentBlock's match we are done.
-			if mcg.CurrentBlock == ncg.CurrentBlock {
+			if lcg.CurrentBlock == ncg.CurrentBlock {
 				return nil
 			}
 			// If the miner's height is greater than the node's we need to
 			// wait a bit longer for them to sync.
-			if mcg.Height > ncg.Height {
-				gwgMiner, errMiner := miner.GatewayGet()
-				gwgNode, errNode := node.GatewayGet()
-				return errors.Compose(fmt.Errorf("the node didn't catch up to the miner's height %v %v but have %v and %v peers respectively",
-					mcg.Height, ncg.Height, len(gwgNode.Peers), len(gwgMiner.Peers)), errMiner, errNode)
+			if lcg.Height != ncg.Height {
+				return errors.New("blockHeight doesn't match")
 			}
 			// If the miner's height is smaller than the node's we need a
 			// bit longer for them to sync.
-			if mcg.Height < ncg.Height {
-				return errors.New("the miner didn't catch up to the node's height")
+			if lcg.CurrentBlock != ncg.CurrentBlock {
+				return errors.New("ids don't match")
```

Comment:
yeah so this could be a perpetual error if you don't verify initially that the block at the highest height is the only block at that height.

Reply:
I've been thinking about that and I think we should keep that as intended behavior. 
`synchronizationCheck` should make sure that all nodes are at the same height but it is not expected to mine a block if that is not the case. 
That means if a test runs into issues because of that it should be handled somewhere else in my opinion.

Reply:
The name `synchronizationCheck` implies that all of the nodes are on the same block - that they are sychronized. If the nodes are not on the same block, it should return an error that the nodes are not on the same block.

Catching this error here instead of earlier means that we have to wait for the full timeout of the `retry` when we could have returned the error immediately.